cmake/common/helpers_common.cmake

```diff
@@ -492,7 +495,27 @@ function(add_obs_plugin target)
     endforeach()
   endif()
 
-  if(found_platform)
+  if(_AOP_NUM_ARCHITECTURES EQUAL 0)
+    set(found_architecture TRUE)
+  else()
+    foreach(architecture IN LISTS _AOP_ARCHITECTURES)
+      if(OS_WINDOWS)
+        if(CMAKE_GENERATOR_PLATFORM STREQUAL "${architecture}")
+          set(found_architecture TRUE)
+        endif()
+      elseif(OS_MACOS)
```

Comment:
You can use `elseif(OS_MACOS AND ${architecture} IN_LIST CMAKE_OSX_ARCHITECTURES)` here - `IN_LIST` will yield true even if `CMAKE_OSX_ARCHITECTURES` is just a single value.

Reply:
I keep a `elseif` and a `if` to not fallback to `CMAKE_SYSTEM_PROCESSOR`.

Reply:
Yeah that's fine, using `IN_LIST` should still be simpler than the regex match.